src/lib/Libattr/attr_func.c

```diff
@@ -1334,3 +1335,370 @@ strtok_quoted(char *source, const char *delimiters)
+ * @return found attrl
+ * @retval NULL if not found
+ */
+
+struct attrl *
+find_attrl(struct attrl *pattrl, char *name, char *resc)
+{
+	while (pattrl) {
+		if (strcmp(name, pattrl->name) == 0) {
+			if (resc) {
+				if (strcmp(resc, pattrl->resource) == 0) {
+					return pattrl;
+				}
+			} else {
+				return pattrl;
+			}
+		}
+		pattrl = pattrl->next;
+	}
+	return ((struct attrl *) 0);
+}
+
+/**
+ *
+ * @brief locate a attribute (attrl) by name/resource and return it's value
+ *
+ * @param[in] pattrl - attrl list to search
+ * @param[in] name   - name of attribute to search for
+ * @param[in] resc   - name of resources to search for (may be NULL of none)
+ *
+ * @return char *
+ * @retval attrl's value
+ * @retval NULL if not found
+ */
+
+char *find_attrlv (struct attrl *pattrl, char *name, char *resc)
+{
+	struct attrl *attrp;
+
+	attrp = find_attrl(pattrl, name, resc);
+
+	if (attrp)
+		return attrp -> value;
+
+	return (char *)NULL;
+}
+
+/**
+ * @brief frees attrl structure
+ *
+ * @param [in] at - attrl to free
+ * @return nothing
+ */
+void
+free_attrl(struct attrl *at)
+{
+	if (at == NULL)
+		return;
+
+	if (at->name != NULL)
+		free(at->name);
+	if (at->resource != NULL)
+		free(at->resource);
+	if (at->value != NULL)
+		free(at->value);
+	free(at);
+}
+
+/**
+ * @brief frees attrl list
+ *
+ * @param[in] at_list - attrl list to free
+ * @return nothing
+ */
+void free_attrl_list(struct attrl *at_list)
+{
+	struct attrl *cur, *tmp;
+	if(at_list == NULL )
+		return;
+
+	for(cur = at_list; cur != NULL; cur = tmp) {
+		tmp = cur->next;
+		free_attrl(cur);
+	}
+
+}
+
+/**
+ * @brief
+ * 	Returns TRUE if the name passed in is an attribute.
+ *
+ * @note
+ * 	This must not be called with object of type MGR_OBJ_SITE_HOOK or MGR_OBJ_PBS_HOOK.
+ *
+ * @param[in]	object - type of object
+ * @param[in]	name  - name of the attribute
+ * @param[in]	attr_type  - type of the attribute
+ *
+ * @eturn int
+ * @retval	TRUE - means if the input is an attribute of the given 'object' type
+ *        	    and 'attr_type'.
+ * @retval	FALSE - otherwise.
+ *
+ */
+int
+is_attr(int object, char *name, int attr_type)
+{
+	static char *svr_public_attrs[] =
+		{
+#include "qmgr_svr_public.h"
+#include "site_qmgr_svr_print.h"
+		NULL
+	};
+
+	static char *svr_readonly_attrs[] =
+		{
+#include "qmgr_svr_readonly.h"
+		NULL
+	};
+
+	static char *que_public_attrs[] =
+		{
+#include "qmgr_que_public.h"
+#include "site_qmgr_que_print.h"
+		NULL
+	};
+
+	static char *que_readonly_attrs[] =
+		{
+#include "qmgr_que_readonly.h"
+		NULL
+	};
+
+	static char *node_public_attrs[] =
+		{
+#include "qmgr_node_public.h"
+#include "site_qmgr_node_print.h"
+		NULL
+	};
+
+	static char *node_readonly_attrs[] =
+		{
+#include "qmgr_node_readonly.h"
+		NULL
+	};
+
+	static char *sched_public_attrs[] =
+		{
+#include "qmgr_sched_public.h"
+#include "site_qmgr_sched_print.h"
+		NULL
+	};
+
+	static char *sched_readonly_attrs[] =
+		{
+#include "qmgr_sched_readonly.h"
+		NULL
+	};
```

Comment:
I decided to re-write is_attr() altogether so that we directly check the master attr lists for the existence of the attribute, and then match the attribute type later. Please let me know what you think.